src/assembler/parser/index.ts

```diff
@@ -27,6 +27,10 @@ export function parse(code: string): ICodeFile {
       name: 'Whitespace',
       pattern: /\s+/
     },
+    {
+      name: 'ProcedureInstructions',
+      pattern: `PROC\|ENDP\|END`
```

Comment:
Split PROC and both END and ENDP into 2 rules since label is allowed before PROC but not the END ones.

Reply:
Have changed this :).